demisto_sdk/commands/content_graph/tests/pack_metadata_graph_test.py

```diff
@@ -292,3 +292,42 @@ def test_pack_metadata_marketplacev2(
     assert metadata_modeling_rule.get("fromversion") == "6.10.0"
     assert metadata_modeling_rule.get("id") == "my_ModelingRule"
     assert metadata_modeling_rule.get("name") == "My Modeling Rule"
+
+
+def test_pack_metadata_dependencies(
+    git_repo: Repo, tmp_path: Path, mocker, monkeypatch
+):
+    """
+    Given:
+        - A repository with two packs, TestPack and TestDependencyPack.
+    When:
+        - In TestPack's pack_metadata is mentioned that it is dependent mandatorily on TestDependencyPack with version 1.2.3 .
```

Comment:
The flow does not consider a case where a pack actually uses an item from TestDependencyPack. This should have created a dependency version higher than 1.2.3, but since we manually added 1.2.3, it will use it.

Reply:
This is not matter whether it uses a content item from the dependency pack or not for the min version, the point here is that in the pack_metadata of TestDependencyPack there is a different current_version than what is mentioned in the TestPack's dependencies minVersion, and yet the resulted metadata has the minVersion from the TestPack's pack_metadata.

Reply:
where I can see the difference? cause in demisto_sdk/commands/content_graph/tests/test_data/pack_metadata.json I see version 1.2.3

Reply:
It creates a generic pack from TestUtils which sets a current_version 1.0.0.
<img width="885" alt="Screenshot 2025-01-30 at 10 32 19" src="https://github.com/user-attachments/assets/d8844186-2e95-4693-b428-4b5dd3a99c6e" />

Reply:
After we create it let's explicitly change that pack current version to 2.0.0. It will highlight the purpose of the feature and test (which is to manually fix the dependency to 1.2.3)